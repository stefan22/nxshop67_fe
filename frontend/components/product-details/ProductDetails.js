import React from 'react'
import Details from './Product.styled'
import { SkeleLoader, withLoading } from '../loader'
const ProdDetails = ({ product }) => (
  <Details animate={{ opacity: 1 }} data-testid="product-details">
    <picture>
      <img
        src={product?.photo.image.publicUrlTransformed}
        alt={product?.name}
        width={640}
        height={725}
      />
    </picture>
    <h2>{product?.name}</h2>
    <p>{product?.price}</p>
    <p>{product?.description}</p>
  </Details>
)

const ProductDetails = withLoading(ProdDetails, SkeleLoader)

export default ProductDetails
