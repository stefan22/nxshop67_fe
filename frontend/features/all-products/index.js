import useProductsPagination from './useProductsPagination'
import paginationQuery from './paginationQuery'

export { useProductsPagination, paginationQuery }
