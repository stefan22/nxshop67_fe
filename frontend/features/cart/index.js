import addToCartMutation from './addToCartMutation'
import removeFromCartMutation from './removeFromCartMutation'

export { addToCartMutation, removeFromCartMutation }
