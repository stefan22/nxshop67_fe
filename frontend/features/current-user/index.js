import useCurrentUser from './useCurrentUser'
import currentUserQuery from './currentUserQuery'

export { useCurrentUser, currentUserQuery }
