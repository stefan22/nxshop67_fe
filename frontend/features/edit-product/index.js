import useEditProduct from './useEditProduct'

export { useEditProduct }
