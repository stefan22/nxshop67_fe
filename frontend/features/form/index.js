import useForm from './useForm'
import signInMutation from './signInMutation'
import signUpMutation from './signUpMutation'
import requestResetMutation from './requestResetMutation'
import resetMutation from './resetMutation'

export { useForm, signInMutation, signUpMutation, requestResetMutation, resetMutation }