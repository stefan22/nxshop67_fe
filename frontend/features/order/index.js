import userOrdersQuery from './userOrdersQuery'
import createOrderMutation from './createOrderMutation'
import singleOrderQuery from './singleOrderQuery'

export { userOrdersQuery, createOrderMutation, singleOrderQuery }
