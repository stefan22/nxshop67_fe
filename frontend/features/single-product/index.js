import useSingleProduct from './useSingleProduct'
import singleProductQuery from './singleProductQuery'

export { useSingleProduct, singleProductQuery }
