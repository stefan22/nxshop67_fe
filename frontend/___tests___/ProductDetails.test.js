/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { singleProductQuery } from '../features/single-product'
import { MockedProvider } from '@apollo/react-testing'
import ProductDetails from '../components/product-details'

const singleProduct = () => ({
  id: 'abc123',
  price: '£270',
  user: null,
  name: 'The New Vans',
  description: 'Off the Wall Packs',
  photo: {
    id: 'abc123',
    altText: 'Off the Wall Packs',
    image: {
      publicUrlTransformed: 'vans.jpg'
    }
  }
})

const vans = singleProduct()

const mocks = [
  {
    request: {
      // product request
      query: singleProductQuery,
      variables: {
        id: 'abc123'
      }
    },
    result: {
      data: {
        Product: vans
      }
    }
  }
]

describe('<DetailsItem />', () => {
  it('renders product details data', async () => {
    const { container, debug } = render(
      <MockedProvider mocks={mocks}>
        <ProductDetails product={vans} />
      </MockedProvider>
    )
    await screen.findByTestId('product-details')
    await screen.findByAltText('The New Vans')
    await screen.findByText('The New Vans')
    // product price
    const productPrice = container.querySelector('p')
    expect(productPrice).toHaveTextContent('£270')
    // product image src
    const imageSrc = container.querySelector('img')
    expect(imageSrc).toBeInTheDocument()
    // product
    expect(container).toMatchSnapshot()
  })
})
