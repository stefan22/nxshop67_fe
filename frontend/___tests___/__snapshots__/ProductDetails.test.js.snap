// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsItem /> renders product details data 1`] = `
<div>
  <div
    class="Productstyled__Details-sc-1ryevin-0 gwBhly"
    data-testid="product-details"
  >
    <picture>
      <img
        alt="The New Vans"
        height="725"
        src="vans.jpg"
        width="640"
      />
    </picture>
    <h2>
      The New Vans
    </h2>
    <p>
      Â£270
    </p>
    <p>
      Off the Wall Packs
    </p>
  </div>
</div>
`;
